ceph/rados/core_workflows.py

```diff
@@ -1746,6 +1751,292 @@ def get_daemon_status(self, daemon_type, daemon_id) -> tuple:
         log.debug(orch_ps_out)
         return orch_ps_out["status"], orch_ps_out["status_desc"]
 
+    def daemon_check_post_tests(
+        self, pre_test_orch_ps: dict, pre_crash_report: list = None
+    ) -> bool:
+        """
+        Method that compares the daemons & their placement beforw and after the test, to check if any daemon is
+        removed, added or placement is changed onto another host.
+
+        Args:
+            pre_test_orch_ps : output of "ceph orch ps" before the tests/ operations
+            pre_crash_report : output of "ceph crash ls" before the tests/ operations
+        Returns:
+            Pass -> True, fail -> False
+        """
+        # Load JSON data
+        orch_ps_data = pre_test_orch_ps
+        cmd = "ceph orch ps"
+        orch_ps_new_data = self.run_ceph_command(cmd=cmd)
+
+        # Function to collect daemon_name and hostname by daemon_type
+        def collect_daemon_info(data):
+            daemon_info = {}
+            for entry in data:
+                daemon_type = entry["daemon_type"]
+                daemon_name = entry["daemon_name"]
+                hostname = entry["hostname"]
+                if daemon_type not in daemon_info:
+                    daemon_info[daemon_type] = {}
+                daemon_info[daemon_type][daemon_name] = hostname
+            return daemon_info
+
+        # Collect daemon information from both JSON structures
+        orch_ps_daemons = collect_daemon_info(orch_ps_data)
+        orch_ps_new_daemons = collect_daemon_info(orch_ps_new_data)
+
+        # Compare daemons and print the results
+        all_match = True
+        for daemon_type, daemons in orch_ps_daemons.items():
+            if daemon_type in orch_ps_new_daemons:
+                for daemon_name, hostname in daemons.items():
```

Comment:
Please add log info of daemon names and hostname

Reply:
added debug logs.